docs/content/docs/build/containers.md

```diff
@@ -19,66 +19,95 @@ In the below scenarios, `client` represents the service-specific client. See the
+[Data modeling]({{< relref "data-modeling.md" >}}).
 
 This example schema defines two initial objects, and declares `SharedCell` and `SharedString` as shared object types that can be dynamically created at runtime.
 
-```typescript
-const schema = {
-    name: "example-container",
+```typescript {linenos=inline}
+const containerSchema = {
     initialObjects: {
-        layout: SharedDirectory,
+        layout: SharedMap,
         text: SharedString
     },
     dynamicObjectTypes: [ SharedCell, SharedString ],
-}
+};
 ```
 
 ### Creating a container
 
 Containers are created from the service-specific client's `createContainer` function. You must provide a config that is specific to the service and a schema object that defines the container schema.
 
-```typescript {hl_lines=[10]}
+```typescript {linenos=inline,hl_lines=[7,8]}
 const schema = {
-    name: "example-container",
     initialObjects: {
         layout: SharedMap,
     },
-}
-const { container, containerServices} =
-    await client.createContainer(/*service config*/, schema);
+};
+
+const { container, services } =
+    await client.createContainer(schema);
+
+const containerId = await container.attach();
 ```
 
-### Loading a container
+### Attaching a container
+
+A newly created container is in a *detached* state. This is the point where you can create initial data to populate your
+shared objects if needed. A detached container is not connected to the Fluid service and no data is shared with other clients.
+
+In order to attach the container to a service, call its `attach` function. Once *attached*, the Fluid container is
+connected to the Fluid service and can be loaded by other clients.
 
-To load the container created in the above section you must provide the service config as well as the exact same schema definition. The same container schema is required on all subsequent loads or the container will not be loaded correctly.
+Note that once attached, a container cannot be detached. Attach is a one-way operation. When loading an existing
+container, the loaded container is always attached.
 
-```typescript {hl_lines=[10]}
+```typescript {linenos=inline,hl_lines=[10]}
 const schema = {
-    name: "example-container",
     initialObjects: {
         layout: SharedMap,
     },
-}
-const { container, containerServices} =
-    await client.getContainer(/*service config*/, schema);
+};
+
+const { container, services } =
+    await client.createContainer(schema);
+
+const containerId = await container.attach();
 ```
 
-### Attaching a container
+### Loading a container
+
+To load a container you must provide the service config as well as the exact same schema definition. The same container
```

Comment:
getContainer always takes an 'id' as the first param right? Not a service config? That config goes into the client creation.

Reply:
```suggestion
To load a container you must provide the `id` of the container to load as well as the exact same schema definition used when creating the container. The same container
```